syncserver-db-common/src/error.rs

```diff
@@ -5,17 +5,17 @@ use http::StatusCode;
 use syncserver_common::{from_error, impl_fmt_display, ReportableError};
 use thiserror::Error;
 
-/// Error specific to any MySQL database backend. These errors are not related to the syncstorage
+/// Error specific to any SQL database backend. These errors are not related to the syncstorage
 /// or tokenserver application logic; rather, they are lower-level errors arising from diesel.
```

Comment:
This comment still stands and I think suggests a clearer name for this: `DieselError`, how does that sound?

Reply:
Honestly, I'm fine either way, but I do appreciate the concept of having a more generic `SqlError` for the reason that @Eragonfr notes. I don't really see us replacing diesel any time soon, but the sources of the error being reported are coming from the underlying SQL system. `DieselError` type should be reserved for errors arising from the diesel system directly (e.g. failure to connect to the data store, or some configuration break).